[dynamo+aten] Enable embedding_bag_byte_rowwise_offsets + meta kernel impl

jfix71 opened the pull request:
Differential Revision: D47007550



cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

gmagogsfm commented on aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp:
```diff
@@ -957,6 +957,29 @@ Tensor embedding_bag_2bit_rowwise_offsets(
+    const Tensor& /* indices */,
+    const c10::optional<Tensor>& offsets_in,
+    const bool /* scale_grad_by_freq */,
+    const int64_t /* mode */,
```
Can you add asserts to make sure the assumptions to made on arguments like `mode` are explicitly enforced?

jfix71 replied:
@gmagogsfm Updated:
* Validated that `mode` and `scale_grad_by_freq` [aren't used at all in the actual op](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp#L889-L890)
* Brought over some checks for `indices` based on ndims/dtypes, and also offsets dtype
* Validated `per_sample_weights_`, `pruned_weights`, and `compressed_indices_mapping` make no difference on the output -- entirely internal to the kernel.

gmagogsfm reviewed (approved):
I don't know the op semantics well, so didn't check the correctness of shape fn.

jfix71 commented on aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp:
```diff
@@ -957,6 +957,45 @@ Tensor embedding_bag_2bit_rowwise_offsets(
+    bool include_last_offset) {
+  TORCH_CHECK(
+      indices.dim() == 1 || indices.dim() == 2,
+      "qembedding/qembedding_bag operator supports 1 or 2d indices, got ",
```
I actually copied this over from elsewhere and thought it'd be good to match 😄 but sure I can change it in this new code.
https://github.com/pytorch/pytorch/blob/02abbb8109a98c2f0a6e01d0ea066731a30acefb/aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp#L534

jfix71 commented:
@pytorchbot rebase master

jfix71 commented:
@pytorchbot rebase -b master

jfix71 committed:
[dynamo+aten] Enable embedding_bag_byte_rowwise_offsets + meta kernel impl (#106105)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/106105

Reviewed By: jaybean-dev, gmagogsfm

Differential Revision: D47007550

fbshipit-source-id: 118c95cbb2de7945dd018da21aa9aec4563c610f
